### ISSUE TITLE
Crash when opening 50MB 108 pdf file

### ISSUE THREAD

**Issue by Terminalator** [pdfarranger/pdfarranger#689]
Describe the bug
----------------

Program crashes on startup when opening a particular file

To Reproduce
------------

1. Open pdfarranger then drag and drop the file into the window or...
2. Right-click file and open with -> pdfarranger

Expected behavior
-----------------

Expect to see a preview of all pages to allow me to edit them

Input files
-----------

One file in particular causes this issue. 108 pages 50MB "multi-session" pdf created in Win7 VM using Fujitsu ix500 and ScanSnap software.

Further Information 
-----------

I get a brief preview of most the pages in a pdfarranger window then it crashes. Terminal output shows this:

`python: ../cairo/src/cairo-error.c:68: _cairo_error: Assertion '_cairo_status_is_error (status)' failed.
Aborted (core dumped)`

Update: atril pdf viewer also crashes with same terminal output text, so this may not be a pdfarranger fault.

Update 2: after extracting some pages in Win7 VM (Adobe Acrobat) and reducing file size to 28.1 MB and 62 pages, the file now opens in pdfarranger without crashing.

System and Versions
-------------------

PDF Arranger-1.8.2
pikepdf-5.1.5.post1
libqpdf-10.6.3
python-cairo 1.21.0-1
Arch Linux fully updated

**Comment by dreua** [comment]
> Update: atril pdf viewer also crashes with same terminal output text, so this may not be a pdfarranger fault.

Yes, it's probably not a bug in PDF Arranger. Without the file I see no way we can investigate this further, unfortunately.
If you want us to have a closer look, you'd need to send it to me or one of the other developers. (My E-Mail should be visible on my Github profile page.)

**Comment by dreua** [comment]
Could you file a bug report at [cairo](https://gitlab.freedesktop.org/cairo/cairo) about this? Feel free to add links in both directions.

**Comment by Terminalator** [comment]
Thanks for taking time to look into this. I've raised an issue with cairo 
[https://gitlab.freedesktop.org/cairo/cairo/-/issues/577](url)

**Comment by dreua** [comment]
Thanks. I think it would be a good idea to include the versions of all Cairo related packages on your system, see my example above. 
I also fixed your markdown link here but I can't do that over there. Just edit and test it until it works, I'm sure you'll figure it out 😉
Finally, please try if the error is reproducible with the file I linked above. If it does, add that link to the issue, developers always appreciate getting all the data they need to reproduce the error.

**Comment by Terminalator** [comment]
Thanks very much for the advice. I've changed my post on gitlab; hope it's better now.

**Comment by psychon** [comment]
> Well, I can reproduce it in the flatpak installation
> [...]
> However, the local RPM install works just fine:

What cairo version is used in the flatpak? Since you write the local RPM works fine and show that it uses 1.17.6-1.fc36 which is only three months old. Could it be that whatever the bug is, it is already fixed in newer cairo versions?

**Comment by dreua** [comment]
Cairo in the Flatpak is part of the GNOME 42 runtime, I haven't found a way to get its version, unfortunately.

**Comment by dreua** [comment]
It would be great if someone could test with both versions and confirm @psychon 's theory.

**Comment by psychon** [comment]
Well... disclaimer: I have no clue about flatpack.

> GNOME 42 runtime,

Google found https://docs.flatpak.org/en/latest/available-runtimes.html which links to https://gitlab.gnome.org/GNOME/gnome-build-meta. [This file](https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/junction.refs) has a reference to `freedesktop-sdk-22.08beta.1-0-g0acffc30b434621d3f5921669eb6c6ae63e7ff2d`. So, apparently the GNOME runtime is based on the Freedesktop one (https://docs.flatpak.org/en/latest/available-runtimes.html says this, too).

Based on that, I found https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/cairo.bst which was last touched two months ago. It also contains lots of references to cairo 1.16. The git history of the file also says that this was last touched two years ago.

So, I guess the flatpack version has a cairo version from 2018. A rough estimate via `git shortlog 1.16.0..origin/master  | grep -E '^ ' | wc -l` says that this is 760 commits behind current cairo master. Compared to 1.17.6, that's still 662 commits behind.

**Comment by kbengs** [comment]
I see the same crash with pdfarranger installed from flatpak. Pdfarranger installed from deb in Ubuntu 22.04 has no issue. Ubuntu has cairo 1.16.0..

**Comment by dreua** [comment]
For the Flatpaks it has been fixed in the mean time, more specifically between those two flatpak-versions:
```
    Commit: 276ae87fe3f526cab755137d630a726bf5020b7a0d2b4735db32ba83e5801f0b #Good
   Subject: Update 2 modules (c78680c2)
      Date: 2022-07-04 21:41:05 +0000

    Commit: 71234b4958007f960bc7136c5031bf370e3ba64132c19e45deeab33acb6514a4 #Bad
   Subject: Add pikepdf dependency: deprecation (6c631ab3)
      Date: 2022-07-03 14:34:02 +0000
```

Unfortunately I still don't have a clue what caused and fixed this.

Could you try again and see if it is fixed for you, too? (Maybe try updating your system first.)

**Comment by dreua** [comment]
Looking at [c78680c2](https://github.com/flathub/com.github.jeromerobert.pdfarranger/commit/c78680c2a0c7354374d5b5e89aaa88e31d15007d) it's probably fixed by Poppler 22.06 -> 22.07 which would be plausible since Poppler is using Cairo as backend.

**Comment by dreua** [comment]
I assume this has been fixed by Poppler Updates and closing this for lack of response. Feel free to reopen if the issue still persists.
